File: php7/src/ChainOfResponsibility/Tests/Agent/TillAgentTest.php
<?php

declare(strict_types = 1);

namespace PetrePatrasc\ChainOfResponsibility\Tests\Agent;


use PetrePatrasc\ChainOfResponsibility\Agent\AbstractLoanAgent;
use PetrePatrasc\ChainOfResponsibility\Agent\TillAgent;

class TillAgentTest extends AbstractLoanAgentTest
{
    /**
     * @var TillAgent
     */
    protected $tillAgent;

    public function setUp()
    {
        parent::setUp();

        $this->tillAgent = new TillAgent();
    }

    /**
     * @inheritDoc
     */
    public function loanAmountDataProvider(): array
    {
        return [
            [10, true],
            [50, true],
            [310.3827, true],
            [999.70, true],
            [4999.99, true],
        ];
    }

    /**
     * @inheritDoc
     */
    protected function getTestedAgent(): AbstractLoanAgent
    {
        return $this->tillAgent;
    }

}

File: java/src/ro/petrepatrasc/chainofresponsability/tests/agent/TillAgentTest.java
package ro.petrepatrasc.chainofresponsability.tests.agent;

import com.tngtech.java.junit.dataprovider.DataProvider;
import com.tngtech.java.junit.dataprovider.DataProviderRunner;
import org.junit.runner.RunWith;
import ro.petrepatrasc.chainofresponsability.agent.AbstractLoanAgent;
import ro.petrepatrasc.chainofresponsability.agent.TillAgent;

@RunWith(DataProviderRunner.class)
public class TillAgentTest extends AbstractLoanAgentTest {

    @Override
    public AbstractLoanAgent getTestedAgent() {
        return new TillAgent();
    }

    @DataProvider
    public static Object[][] loanAmountDataProvider() {
        return new Object[][]{
                {312.17, true},
                {10.38, true},
                {512.75, true},
                {3985.48, true},
                {4999.99, true},
        };
    }
}
